CMakeLists.txt

```diff
@@ -1,32 +1,47 @@
 cmake_minimum_required(VERSION 3.16)
-project(Noxis)
+project(Noxis C)
 
-set(CMAKE_C_FLAGS "-Wall -Wpedantic -Wextra -Werror")
```

Comment:
This is a portability issue. These flags won't work for every compiler, also this is a global definition. Prefer target definitions.